tests/testConstantSettings.php

```diff
@@ -0,0 +1,189 @@
+	 */
+	public function testSet() {
+		$opt_name       = 'domain';
+		$expected_val_1 = rand();
+		$expected_val_2 = rand();
+		$opts           = new WP_Auth0_Options();
+
+		// Test that a basic set without DB update works.
+		$result = $opts->set( $opt_name, $expected_val_1, false );
+		$this->assertTrue( $result );
+		$this->assertEquals( $expected_val_1, $opts->get( $opt_name ) );
+		$db_options = get_option( $opts->get_options_name() );
+		$this->assertNotEquals( $expected_val_1, $db_options[ $opt_name ] );
+
+		// Test that a basic set with DB update works.
+		$result = $opts->set( $opt_name, $expected_val_2, true );
+		$this->assertTrue( $result );
+		$this->assertEquals( $expected_val_2, $opts->get( $opt_name ) );
+		$db_options = get_option( $opts->get_options_name() );
+		$this->assertEquals( $expected_val_2, $db_options[ $opt_name ] );
+	}
+
+	/**
+	 * Test that options cannot be set when a constant is present.
+	 */
+	public function testSetWithConstant() {
+		$opt_name      = 'domain';
+		$expected_val  = rand();
+		$opts          = new WP_Auth0_Options();
+		$constant_name = $opts->get_constant_name( $opt_name );
+
+		// Set a constant and make sure it works.
+		$this->assertNull( $opts->get_constant_val( $opt_name ) );
+		define( $constant_name, $expected_val );
+		$this->assertEquals( $expected_val, $opts->get_constant_val( $opt_name ) );
+
+		// Try to set an option with the constant set.
+		$result = $opts->set( $opt_name, rand(), false );
+		$this->assertFalse( $result );
+		$this->assertNotEquals( $expected_val, $opts->get( $opt_name ) );
+	}
+
+	/**
+	 * Test that commonly-overridden settings will show a notice.
+	 */
+	public function testConstantSettingNoticeBasic() {
+		$opts  = new WP_Auth0_Options();
+		$admin = new WP_Auth0_Admin_Basic( $opts );
+
+		$fields = [
+			[
+				'opt_name'        => 'domain',
+				'label_for'       => 'wpa0_domain',
+				'render_function' => 'render_domain',
+			],
+			[
+				'opt_name'        => 'client_id',
+				'label_for'       => 'wpa0_client_id',
+				'render_function' => 'render_client_id',
+			],
+			[
+				'opt_name'        => 'client_secret',
+				'label_for'       => 'wpa0_client_secret',
+				'render_function' => 'render_client_secret',
+			],
+			[
+				'opt_name'        => 'auth0_app_token',
+				'label_for'       => 'wpa0_auth0_app_token',
+				'render_function' => 'render_auth0_app_token',
+			],
+		];
+
+		foreach ( $fields as $field ) {
+			$constant_name = self::DEFAULT_CONSTANT_PREFIX . strtoupper( $field['opt_name'] );
+			$override_val  = self::FILTER_TEST_STRING . rand();
+			define( $constant_name, $override_val );
+
+			ob_start();
+			$admin->{$field['render_function']}( $field );
+			$field_html = ob_get_clean();
+
+			$input = $this->getDomListFromTagName( $field_html, 'input' );
+			$this->assertTrue( $input->item( 0 )->hasAttribute( 'disabled' ) );
+			$this->assertNotFalse( strpos( $field_html, self::CONSTANT_NOTICE_TEXT ) );
```

Comment:
`assertNotFalse` vs `assertTrue` ?

Reply:
`strpos()` returns the position of the string as an integer (could be `0`) or `FALSE` if it's not found. I want to check that the string is there so I want to assert that it's not false. 

That said, there is an `assertContains` that works better here so I switched to that.